app/src/main/java/com/duckduckgo/app/bookmarks/ui/BookmarksActivity.kt

```diff
@@ -274,7 +280,12 @@ class BookmarksActivity : DuckDuckGoActivity() {
         searchBar.onAction {
             when (it) {
                 is SearchBar.Action.PerformUpAction -> hideSearchBar()
-                is SearchBar.Action.PerformSearch -> listener.onQueryTextChange(it.searchText)
+                is SearchBar.Action.PerformSearch -> {
+                    searchJob += lifecycle.coroutineScope.launch {
```

Comment:
Have you considered debouncing in the listener itself? It has a reference to the `viewModel`, so you could use `viewModel.viewModelScope.launch` and avoid canceling the job in the `Activity`.

It would also remove the `DEBOUNCE_PERIOD ` from here

Reply:
Good suggestion, thanks! Updated :+1: